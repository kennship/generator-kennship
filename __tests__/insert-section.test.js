const insertSection = require('../generators/util/insert-section');

describe('util/insert-section', () => {
  it('should insert a new section', async () => {
    const markdown = `# generator-kennship

> Common tasks for new projects

[![NPM version][npm-image]][npm-url]
[![Build Status][travis-image]][travis-url]
[![Coverage percentage][coveralls-image]][coveralls-url]

## Installation

Details on installing.

## Usage

Details on usage.

[npm-image]: https://badge.fury.io/js/generator-kennship.svg
[npm-url]: https://npmjs.org/package/generator-kennship
[coveralls-image]: https://coveralls.io/repos/ryaninvents/generator-kennship/badge.svg
[coveralls-url]: https://coveralls.io/r/ryaninvents/generator-kennship
`;
    const newSection = `## New section\n\nMore details for you.`;
    const updatedReadme = await insertSection(markdown, newSection, {
      priority: 50,
    });

    expect(updatedReadme).toMatchSnapshot();
  });

  it('should behave well in a mostly-empty README', async () => {
    const markdown = `# generator-kennship

> Common tasks for new projects

`;
    const newSection = `## New section\n\nMore details for you.`;
    const updatedReadme = await insertSection(markdown, newSection, {
      priority: 50,
    });

    expect(updatedReadme).toMatchSnapshot();
  });
});
