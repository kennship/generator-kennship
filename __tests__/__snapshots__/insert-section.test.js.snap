// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/insert-section should behave well in a mostly-empty README 1`] = `
"# generator-kennship

> Common tasks for new projects

## New section

More details for you.

"
`;

exports[`util/insert-section should insert a new section 1`] = `
"# generator-kennship

> Common tasks for new projects

[![NPM version][npm-image]][npm-url]
[![Build Status][travis-image]][travis-url]
[![Coverage percentage][coveralls-image]][coveralls-url]

## Installation

Details on installing.

## New section

More details for you.


## Usage

Details on usage.

[npm-image]: https://badge.fury.io/js/generator-kennship.svg

[npm-url]: https://npmjs.org/package/generator-kennship

[coveralls-image]: https://coveralls.io/repos/ryaninvents/generator-kennship/badge.svg

[coveralls-url]: https://coveralls.io/r/ryaninvents/generator-kennship
"
`;
